File: app/src/main/java/com/callmangement/EHR/ehrActivities/BaseActivity.java
package com.callmangement.EHR.ehrActivities;

import androidx.appcompat.app.AppCompatActivity;

public class BaseActivity extends AppCompatActivity {

    // Used to load the 'e_hr' library on application startup.
    static {
        System.loadLibrary("e_hr");
    }
     public static native String BaseUrl();
    // Login
    public native String Login();
    public native String LoginQueryParams();

    // Logout
    public native String Logout();
    public native String LogoutQueryParams();

    // CheckVersion
    public native String CheckVersion();
    public native String CheckVersionQueryParams();

    // DistictList
    public native String DistictList();

    // TehsilListByDistict
    public native String TehsilListByDistict();
    public native String TehsilListByDistictQueryParams();

    // SEUserListByDistict
    public native String SEUserListByDistict();
    public native String SEUserListByDistictQueryParams();

    // CreateCamp
    public native String CreateCamp();
    public native String CreateCampQueryParams();

    // GetCampList
    public native String GetCampList();
    public native String GetCampListQueryParams();

    // OrganiseACamp
    public native String OrganiseACamp();
    public native String OrganiseACampQueryParams();

    // GetDealersByBlock
    public native String GetDealersByBlock();
    public native String GetDealersByBlockQueryParams();

    // UploadDocumentOfDailyWorkReport
    public native String UploadDocumentOfDailyWorkReport();
    public native String UploadDocumentOfDailyWorkReportQueryParams();

    // GetCampDocList
    public native String GetCampDocList();
    public native String GetCampDocListQueryParams();

    // DeleteACamp
    public native String DeleteACamp();
    public native String DeleteACampQueryParams();

    // CreateSurveyForm
    public native String CreateSurveyForm();
    public native String CreateSurveyFormQueryParams();

    // EditSurveyForm
    public native String EditSurveyForm();
    public native String EditSurveyFormQueryParams();

    // GetSurveyFormList
    public native String GetSurveyFormList();
    public native String GetSurveyFormListQueryParams();

    // DeleteASurveyForm
    public native String DeleteASurveyForm();
    public native String DeleteASurveyFormQueryParams();

    // UploadDocumentOfSurveyFormReport
    public native String UploadDocumentOfSurveyFormReport();
    public native String UploadDocumentOfSurveyFormReportQueryParams();

    // GetSurveyFormDocList
    public native String GetSurveyFormDocList();
    public native String GetSurveyFormDocListQueryParams();

    // GetAttendanceList
    public static native String GetAttendanceList();
    public static native String GetAttendanceListQueryParams();

    // MArkAttendance
    public native String MArkAttendance();
    public native String MArkAttendanceQueryParams();
}

File: app/src/main/java/com/callmangement/ehr/ehrActivities/BaseActivity.kt
package com.callmangement.ehr.ehrActivities

import androidx.appcompat.app.AppCompatActivity

open class BaseActivity : AppCompatActivity() {
    // Login
    external fun Login(): String?
    external fun LoginQueryParams(): String?

    // Logout
    external fun Logout(): String?
    external fun LogoutQueryParams(): String?

    // CheckVersion
    external fun CheckVersion(): String?
    external fun CheckVersionQueryParams(): String?

    // DistictList
    external fun DistictList(): String?

    // TehsilListByDistict
    external fun TehsilListByDistict(): String?
    external fun TehsilListByDistictQueryParams(): String?

    // SEUserListByDistict
    external fun SEUserListByDistict(): String?
    external fun SEUserListByDistictQueryParams(): String?

    // CreateCamp
    external fun CreateCamp(): String?
    external fun CreateCampQueryParams(): String?

    // GetCampList
    external fun GetCampList(): String?
    external fun GetCampListQueryParams(): String?

    // OrganiseACamp
    external fun OrganiseACamp(): String?
    external fun OrganiseACampQueryParams(): String?

    // GetDealersByBlock
    external fun GetDealersByBlock(): String?
    external fun GetDealersByBlockQueryParams(): String?

    // UploadDocumentOfDailyWorkReport
    external fun UploadDocumentOfDailyWorkReport(): String?
    external fun UploadDocumentOfDailyWorkReportQueryParams(): String?

    // GetCampDocList
    external fun GetCampDocList(): String?
    external fun GetCampDocListQueryParams(): String?

    // DeleteACamp
    external fun DeleteACamp(): String?
    external fun DeleteACampQueryParams(): String?

    // CreateSurveyForm
    external fun CreateSurveyForm(): String?
    external fun CreateSurveyFormQueryParams(): String?

    // EditSurveyForm
    external fun EditSurveyForm(): String?
    external fun EditSurveyFormQueryParams(): String?

    // GetSurveyFormList
    external fun GetSurveyFormList(): String?
    external fun GetSurveyFormListQueryParams(): String?

    // DeleteASurveyForm
    external fun DeleteASurveyForm(): String?
    external fun DeleteASurveyFormQueryParams(): String?

    // UploadDocumentOfSurveyFormReport
    external fun UploadDocumentOfSurveyFormReport(): String?
    external fun UploadDocumentOfSurveyFormReportQueryParams(): String?

    // GetSurveyFormDocList
    external fun GetSurveyFormDocList(): String?
    external fun GetSurveyFormDocListQueryParams(): String?

    // MArkAttendance
    external fun MArkAttendance(): String?
    external fun MArkAttendanceQueryParams(): String?

    companion object {
        // Used to load the 'e_hr' library on application startup.
        init {
            System.loadLibrary("e_hr")
        }

        @JvmStatic
        external fun BaseUrl(): String?

        // GetAttendanceList
        @JvmStatic
        external fun GetAttendanceList(): String?
        @JvmStatic
        external fun GetAttendanceListQueryParams(): String?
    }
}